images.go

```diff
@@ -49,11 +49,32 @@ type Image struct {
 	Flags map[string]interface{} `json:"flags,omitempty"`
 }
 
+// ROImageStore provides bookkeeping for information about Images.
+type ROImageStore interface {
+	ROFileBasedStore
+	ROMetadataStore
+	ROBigDataStore
+
+	// Exists checks if there is an image with the given ID or name.
+	Exists(id string) bool
```

Comment:
The interfaces limit which methods are available through a value of that interface type.  While a value of type `FlaggableStore` doesn't have these methods, an `ImageStore` value, which is of a type that embeds both `ROImageStore` and `FlaggableStore`, includes both (and more), and that's what we actually use.
They're sliced up like this to avoid having any given method appear more than once in an interface, whether directly or through embedding.  The compiler rejected that kind of duplication as an error when I first tried it.
The method implementations receive the underlying object type, so they're free to call any method that the object implements.
If we implemented read-only and read-write stores as different object types in addition to different interface types, we wouldn't need to check for read-only status in several of the write methods.  We're adding those checks just to be safe in case someone decides to try using a type assertion to convert a read-only interface type to a read-write interface.  The object that underlies a read-only interface value also provides all of the methods of the read-write interface, so it wouldn't cause an error.